src/main/java/tempest/ui/pages/GoalEntryPage.java

```diff
@@ -0,0 +1,190 @@
+package tempest.ui.pages;
+
+import tempest.Module;
+import tempest.State;
+import tempest.ui.ErrorMessage;
+import tempest.ui.GUIManager;
+import tempest.ui.components.ActionButtonPanel;
+import tempest.ui.components.ClearButton;
+import tempest.ui.components.ModuleDropDown;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.time.Duration;
+import java.util.Objects;
+
+public class GoalEntryPage extends Page implements InputPage{
+
+    private final State state;
+    private final ModuleDropDown moduleDropDown = new ModuleDropDown();
+    private final ActionButtonPanel actionButtonPanel;
+    private final ErrorMessage errorMessage = new ErrorMessage();
+
+    private JComboBox<Object> dropDown;
+    private JTextField hoursInput;
+    private JTextField minutesInput;
+    private JButton enterButton;
+
+    public GoalEntryPage(State state, GUIManager guiManager){
+        super(guiManager);
+        this.state = state;
+
+        this.actionButtonPanel = new ActionButtonPanel(guiManager, this);
+        setupUI();
+    }
+
+    @Override
+    public String getName() {
+        return PageNames.GOAL_ENTRY;
+    }
+
+    private void setupUI() {
+        enterButton = (JButton) actionButtonPanel.getComponent(1);
+
+        JPanel inputPanel = new JPanel();
+
+        dropDown = moduleDropDown.getModuleDropDown();
+        inputPanel.add(dropDown);
+
+        hoursInput = new JTextField(2);
+        JLabel hoursInputLabel = new JLabel("Hours");
+
+        minutesInput = new JTextField(2);
+        JLabel minutesInputLabel = new JLabel("Minutes");
+
+        JPanel hoursPanel = new JPanel();
+        hoursPanel.add(hoursInputLabel);
+        hoursPanel.add(hoursInput);
+
+        JPanel minutesPanel = new JPanel();
+        minutesPanel.add(minutesInputLabel);
+        minutesPanel.add(minutesInput);
+
+        JPanel timeInputPanel = new JPanel();
+        timeInputPanel.add(hoursPanel);
+        timeInputPanel.add(minutesPanel);
+        timeInputPanel.setLayout(new FlowLayout());
+
+        inputPanel.add(timeInputPanel);
+
+        ClearButton clearButton = new ClearButton(this);
+        inputPanel.add(clearButton);
+
+        this.add(inputPanel);
+        this.add(actionButtonPanel);
+        this.setLayout(new BoxLayout(this, BoxLayout.Y_AXIS));
+    }
+
+    @Override
+    public void clearInput() {
+        hoursInput.setText(""); // Clearing hours textbox
+        minutesInput.setText(""); // Clearing minutes textbox
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        Object source = e.getSource();
+
+        if (source == enterButton) {
+            handleSettingGoal();
+        }
+    }
+
+    private void handleSettingGoal(){
+        if (dropDown.getItemCount() < 1) { // No modules created, so can't add a session
```

Comment:
`// No modules created, so can't add a **session**`

Reply:
It is changed.